velox/connectors/hive/storage_adapters/abfs/tests/AbfsFileSystemTest.cpp

```diff
@@ -15,29 +15,86 @@
  */
 
 #include "velox/connectors/hive/storage_adapters/abfs/AbfsFileSystem.h"
+#include <azure/storage/files/datalake.hpp>
 #include "gtest/gtest.h"
 #include "velox/common/base/tests/GTestUtils.h"
 #include "velox/common/file/File.h"
 #include "velox/common/file/FileSystems.h"
 #include "velox/connectors/hive/FileHandle.h"
 #include "velox/connectors/hive/HiveConfig.h"
 #include "velox/connectors/hive/storage_adapters/abfs/AbfsReadFile.h"
+#include "velox/connectors/hive/storage_adapters/abfs/AbfsWriteFile.h"
 #include "velox/connectors/hive/storage_adapters/abfs/tests/AzuriteServer.h"
 #include "velox/exec/tests/utils/PortUtil.h"
 #include "velox/exec/tests/utils/TempFilePath.h"
 
 #include <atomic>
+#include <filesystem>
 #include <random>
 
 using namespace facebook::velox;
-
+using namespace facebook::velox::filesystems::abfs;
+using namespace Azure::Storage::Files::DataLake;
 using ::facebook::velox::common::Region;
 
 constexpr int kOneMB = 1 << 20;
 static const std::string filePath = "test_file.txt";
 static const std::string fullFilePath =
     facebook::velox::filesystems::test::AzuriteABFSEndpoint + filePath;
 
+// A mocked blob storage file client backend with local file store.
+class MockBlobStorageFileClient : public IBlobStorageFileClient {
+ public:
+  MockBlobStorageFileClient() {
+    char tempFileName[] = "/tmp/velox_abfs_test_XXXXXX";
```

Comment:
Use `createTempDirectory` API.

Reply:
hey @majetideepak  this is used to create a temp file not a temp directory

Reply:
There is a `TempFilePath` API as well.
`/velox/exec/tests/utils/TempFilePath.h`

Reply:
updated. thanks @majetideepak